src/containers/Elements/FormWizard.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from 'network-canvas-ui';
+import { Pips } from '../../components/Elements';
+import { Form } from '.';
+
+class FormWizard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      fieldIndex: 0,
+    };
+  }
+
+  onSubmit = (formData, dispatch, form) => {
+    if (this.shouldShowNextField()) {
+      this.nextField();
+      return;
+    }
+    this.props.onSubmit(formData, dispatch, form);
+    this.setState({
+      fieldIndex: 0,
+    });
+  }
+
+  nextField = () => {
+    const count = this.props.fields.length;
+    this.setState({
+      fieldIndex: (this.state.fieldIndex + 1 + count) % count,
+    });
+  };
+
+  previousField = () => {
+    const count = this.props.fields.length;
+    this.setState({
+      fieldIndex: (this.state.fieldIndex - 1 + count) % count,
+    });
+  };
+
+  shouldShowNextField = () => {
+    const showingLastField = this.state.fieldIndex === this.props.fields.length - 1;
+    return !showingLastField;
+  }
+
+  render() {
+    const { fields, onSubmit, ...rest } = this.props;
+
+    const extraProps = {};
+    extraProps.fields = [fields[this.state.fieldIndex]];
+    extraProps.onSubmit = this.onSubmit;
+    if (this.shouldShowNextField()) {
+      extraProps.controls = [(
```

Comment:
Just a thought, but this seems all seems little opaque?

What about something more like:
```javascript
currentField() {
  return [this.props.fields[this.state.fieldIndex]];
}

render() {
  const nextButton = (
    <button key="next" className="form__next-button" aria-label="Submit">
      <Icon name="form-arrow-right" />
    </button>
  );

  const formProps = {
    ...this.props,
    fields: this.currentField(),
    onSubmit: this.onSubmit,
  }

  // ...

  <Form
     {...formProps}
     controls={this.shouldShowNextButton() ? [nextButton] : undefined}
  />
}
```